### PR TITLE
Escape and enclose containerEnv in quotes when writing to Dockerfile

### PR DESCRIPTION
**Pull request by joshspicer**
ref: https://github.com/devcontainers/cli/issues/454

### CODE REVIEW

**Review by samruddhikhandale — DISMISSED** [review]
Nice!

File: src/spec-node/imageMetadata.ts [hunk 1]
```diff
@@ -494,8 +494,11 @@ export function getContainerEnvMetadata(containerEnv: Record<string, string> | u
 	if (!containerEnv) {
 		return '';
 	}
-
 	const keys = Object.keys(containerEnv);
-	const concatenatedEnv = keys.map(k => `ENV ${k}=${containerEnv![k]}`).join('\n');
+	// https://docs.docker.com/engine/reference/builder/#envs
+	const concatenatedEnv = keys.map(k => `ENV ${k}="${containerEnv[k]
+		.replace(/(?=["\\$])/g, '\\') // escape " and \
```
Comment (jkeech) [inline src/spec-node/imageMetadata.ts:500]: What's the purpose of matching the `$` character? Can you add a test to help explain and cover this scenario? Same thing for the backslash literal `\`.

File: src/spec-node/imageMetadata.ts [hunk 2]
```diff
@@ -494,8 +494,11 @@ export function getContainerEnvMetadata(containerEnv: Record<string, string> | u
 	if (!containerEnv) {
 		return '';
 	}
-
 	const keys = Object.keys(containerEnv);
-	const concatenatedEnv = keys.map(k => `ENV ${k}=${containerEnv![k]}`).join('\n');
+	// https://docs.docker.com/engine/reference/builder/#envs
+	const concatenatedEnv = keys.map(k => `ENV ${k}="${containerEnv[k]
+		.replace(/(?=["])/g, '\\') // escape double quotes
```
Comment (chrmarti) [inline src/spec-node/imageMetadata.ts:500]: Would backslashes also require escaping?

**Review by chrmarti — CHANGES_REQUESTED** [review]

File: src/spec-node/imageMetadata.ts (hunk 2 above)
Comment (jkeech) [reply src/spec-node/imageMetadata.ts:500]: Yes, I think we need to escape backslashes as well. Good call.

Comment (jkeech) [reply src/spec-node/imageMetadata.ts:500]: Actually I think the original regex was probably correct: `replace(/(?=["\\$])/g, '\\')`. Looks like we need to escape `$` as well.

**Review by chrmarti — APPROVED** [review]
👍

**Pull request merged**